plugins/outputs/mqtt/topic_name_generator.go

```diff
@@ -4,21 +4,23 @@ import (
 	"fmt"
 	"strings"
 	"text/template"
+	"time"
 
 	"github.com/Masterminds/sprig/v3"
 
 	"github.com/influxdata/telegraf"
 )
 
 type TopicNameGenerator struct {
-	Hostname    string
 	TopicPrefix string
-	PluginName  string
-	metric      telegraf.Metric
+	metric      telegraf.TemplateMetric
 	template    *template.Template
 }
 
 func NewTopicNameGenerator(topicPrefix, topic string) (*TopicNameGenerator, error) {
+	topic = strings.ReplaceAll(topic, ".PluginName", ".Name")
+	topic = strings.ReplaceAll(topic, ".Hostname", `.Tag "host"`)
```

Comment:
Please use a regexp for matching to not match strings like `foo.Hostname.bar` e.g.
```regex
{{\s*\.Hostname\s*}}
```
Same for other occurrences.

Reply:
Agreed, however it would be better to use `\B\.Hostname\b` instead?

[Examples](https://regex101.com/r/z4bjM5/1)

Reply:
I don't think we can leave out the curly brackets...

Reply:
Then you don't support for example the sprig functions like for example `{{ upper .Hostname }}`.

Or do you mean that we should not replace a literal `.Hostname` somewhere in the string?

Reply:
Hmmm I tested `{{ upper .Hostname }}` and it is matched by the expression I proposed (`{{.*\B\.Hostname\b[^}]*}}`)...  Of course you need to use the expressions with groups i.e. `({{.*\B)\.Hostname(\b[^}]*}})` and then do `$1.Tag "host"$2`...

Reply:
See https://goplay.space/#AVELIyF-1Hv

Reply:
To be clear, also do this for `.PluginName`?